xskillscore/core/comparative.py

```diff
@@ -0,0 +1,151 @@
+import warnings
+
+import scipy.stats as st
+import xarray as xr
+
+
+def sign_test(
+    observation,
+    forecast1,
+    forecast2,
+    dim=None,
+    alpha=0.05,
+    metric=None,
+    orientation='negative',
+):
+    """
+        Returns the Delsole and Tippett sign test over the given time dimension.
+
+        Parameters
+        ----------
+        observation : xarray.Dataset or xarray.DataArray or None
+            observation to be compared to both forecasts.
+            If ``None``, then assume that forecast1 and forecast2 have already been
+            compared to observation. If metric is None, this assumes forecasts to be
+            already compared to observation before and ignores observation. Please
+            adjust ``orientation`` accordingly.
+        forecast1 : xarray.Dataset or xarray.DataArray
+            forecast1 to be compared to observation
+        forecast2 : xarray.Dataset or xarray.DataArray
+            forecast2 to be compared to observation
+        dim : str
+            time dimension of dimension over which to compute the random walk.
+            This dimension is not reduced, unlike in other xskillscore functions.
+        alpha : float
+            significance level for random walk.
+        metric : callable, optional
+            metric to compare forecast# with observation if metric is not None. If
+            metric is None, assume that forecast# have been compared observation before
+            using ``sign_test`` and adjust ``orientation``.
+            Use ``metric=categorical``, if the winning forecast should only be rewarded
+            a point if it exactly equals the observation. Also allows strings to be
+            convered to ``xskillscore.{metric}``. Defaults to None.
+        orientation : str
+            Which skill values correspond to better skill? Smaller values (negative) or
+            Larger values (positive). Defaults to 'negative'. Ignored if
+            ``metric== categorical``.
```

Comment:
Maybe clearly put: "One of ['positive', 'negative']". Also, should the default be `negative`? I would argue we should force this as a keyword so they are aware of this. A number of metrics require smaller (negative) scores, but many also require higher scores to be better.

Reply:
most metrics that can be used for the test, e.g. not correlation, are negative.